src/stores/courseData.js

```diff
@@ -287,28 +295,45 @@ const store = new Vuex.Store({
 });
 
 function getMatchingAttributes (gir, hass, ci) {
-  const matchingClasses = store.state.subjectsInfo.filter(function (subject) {
-    if (gir !== undefined && subject.gir_attribute !== gir) {
-      return false;
-    }
-    if (hass !== undefined && subject.hass_attribute !== hass) {
-      return false;
+  if (store.state.fullSubjectInfo) {
+    const matchingClasses = store.state.subjectsInfo.filter(function (subject) {
+      if (store.state.fullSubjectsInfo === false) {
+        return false;
+      } else {
+        if (gir !== undefined && subject.gir_attribute !== gir) {
+          return false;
+        }
+        if (hass !== undefined && subject.hass_attribute !== hass) {
+          return false;
+        }
+        return !(ci !== undefined && subject.communication_requirement !== ci);
+      }
     }
-    return !(ci !== undefined && subject.communication_requirement !== ci);
-  });
-  const totalObject = matchingClasses.reduce(function (accumObject, nextClass) {
-    return {
-      offered_spring: accumObject.offered_spring || nextClass.offered_spring,
-      offered_summer: accumObject.offered_summer || nextClass.offered_summer,
-      offered_IAP: accumObject.offered_IAP || nextClass.offered_IAP,
-      offered_fall: accumObject.offered_fall || nextClass.offered_fall,
-      in_class_hours: accumObject.in_class_hours + (nextClass.in_class_hours !== undefined ? nextClass.in_class_hours : 0),
-      out_of_class_hours: accumObject.out_of_class_hours + (nextClass.out_of_class_hours !== undefined ? nextClass.out_of_class_hours : 0)
-    };
-  }, { offered_spring: false, offered_summer: false, offered_IAP: false, offered_fall: false, in_class_hours: 0, out_of_class_hours: 0 });
-  totalObject.in_class_hours /= matchingClasses.length;
-  totalObject.out_of_class_hours /= matchingClasses.length;
-  return totalObject;
+    );
+    const totalObject = matchingClasses.reduce(function (accumObject, nextClass) {
+      if (store.state.fullSubjectsInfo === false) {
+        return {
+          offered_spring: accumObject.offered_spring || nextClass.offered_spring,
+          offered_IAP: accumObject.offered_IAP || nextClass.offered_IAP,
+          offered_fall: accumObject.offered_fall || nextClass.offered_fall
```

Comment:
Since `undefined || undefined` is `undefined` you may not actually have to duplicate this and have the if statement. The only way I can see this being a problem is if other code checks for `offered_summer in obj` instead of `obj.offered_summer === undefined`